### ISSUE TITLE
[Bug]: fix huggingface streaming error 

### ISSUE THREAD

**Issue by krrishdholakia** [BerriAI/litellm#2180]
### What happened?
```python
{
    "model": "MiquMaid-v2-70B",
    "prompt": "Hello",
    "top_p": 0,
    "stream": true
}
```
streaming returns 200, not 400 

### Twitter / LinkedIn details

cc: @NiklasWilson

**Comment by ghost** [comment]
How openai deals with streaming errors. 
<img width="801" alt="Screenshot 2024-03-04 at 8 04 48 AM" src="https://github.com/BerriAI/litellm/assets/17561003/7459a031-c9dd-498f-aa34-22be278c450a">

**Comment by ghost** [comment]
How we raise errors for non-openai providers 

<img width="1085" alt="Screenshot 2024-03-04 at 8 09 26 AM" src="https://github.com/BerriAI/litellm/assets/17561003/4999e280-1666-4c20-8641-87a56016bf7d">

**Comment by ghost** [comment]
Looks like huggingface itself returns a 200 response code for incorrect values 

<img width="611" alt="Screenshot 2024-03-04 at 8 19 18 AM" src="https://github.com/BerriAI/litellm/assets/17561003/1bd21b8a-5602-4034-8a0e-2672bee837b5">

**Comment by ghost** [comment]
returns a 200 status code, and the chunk contains the error 
<img width="1607" alt="Screenshot 2024-03-04 at 8 20 32 AM" src="https://github.com/BerriAI/litellm/assets/17561003/d230fa74-38db-4637-901e-9573f2d1bd22">

**Comment by ghost** [comment]
Getting somewhere
<img width="908" alt="Screenshot 2024-03-04 at 8 23 03 AM" src="https://github.com/BerriAI/litellm/assets/17561003/b70fb1d4-9e7d-49b4-8010-18df2b5464df">

**Comment by ghost** [comment]
correctly able to receive the error on the proxy from the sdk 
<img width="1435" alt="Screenshot 2024-03-04 at 8 24 14 AM" src="https://github.com/BerriAI/litellm/assets/17561003/27303519-b2cd-482c-8b1b-17087bde370f">

**Comment by ghost** [comment]
i think this should solve it? @NiklasWilson
<img width="1186" alt="Screenshot 2024-03-04 at 8 28 45 AM" src="https://github.com/BerriAI/litellm/assets/17561003/a6ea2a58-31e0-4b96-9903-6119904e6664">


will test with openai client sdk to confirm

**Comment by NiklasWilson** [comment]
@krrishdholakia That looks like it should work, I assume changing the request status code from 200 to 500 is not possible?

**Comment by ghost** [comment]
I believe both openai and hf start the stream and then raise the error as a chunk in the stream

**Comment by ghost** [comment]
Testing with openai sdk. 

## OpenAI Error (wrong input params to gpt-3.5-turbo)
<img width="1606" alt="Screenshot 2024-03-04 at 9 04 17 AM" src="https://github.com/BerriAI/litellm/assets/17561003/c4ebaf75-5f4c-42d9-9f8b-01f81db11b1c">

## OpenAI Error (wrong input params to hf) 
<img width="1457" alt="Screenshot 2024-03-04 at 9 05 54 AM" src="https://github.com/BerriAI/litellm/assets/17561003/2a8e12f7-b858-430b-950e-aef0aff6e22d">

**Comment by ghost** [comment]
I'm wrong. OpenAI does raise the error before starting the stream. 

Maybe we can check the first hf chunk before starting the stream?

**Comment by ghost** [comment]
Got it working exactly like openai. No 200 returned. will return the error immediately. @NiklasWilson

<img width="1604" alt="Screenshot 2024-03-04 at 9 29 21 AM" src="https://github.com/BerriAI/litellm/assets/17561003/715638a4-7807-44cb-918d-9cf7b51ece3a">

**Comment by ghost** [comment]
Fix live soon in v`1.28.14` - https://github.com/BerriAI/litellm/commits/main/
